compiler/src/test/java/io/neow3j/compiler/ContractEventsTest.java

```diff
@@ -189,4 +193,136 @@ public static void fireEvent() {
+                () -> new Compiler().compile(EventParameterNamesContract_TooMany.class.getName()));
+        assertThat(thrown.getMessage(), is("Too many parameter names provided for event Transfer."));
+    }
+
+    static class EventParameterNamesContract_TooMany {
+        @DisplayName("Transfer")
+        @EventParameterNames({"Amount", "Sender", "Hash", "Data", "Success"})
+        public static Event3Args<Integer, Hash160, Hash256> onTransfer;
+    }
+
+    @Test
+    public void testEventParameterNamesWithTooManyNamesButNoneExpected() {
+        CompilerException thrown = assertThrows(CompilerException.class,
+                () -> new Compiler().compile(EventParameterNamesContract_TooManyWhenNoneExpected.class.getName()));
+        assertThat(thrown.getMessage(), is("Too many parameter names provided for event Transfer."));
+    }
+
+    static class EventParameterNamesContract_TooManyWhenNoneExpected {
+        @DisplayName("Transfer")
+        @EventParameterNames({"Anything"})
+        public static io.neow3j.devpack.events.Event onTransfer;
+    }
+
+    @Test
+    public void testEventParameterNamesWithNoNameButNoneExpected() throws IOException {
+        CompilationUnit compUnit = new Compiler().compile(
+                EventParameterNamesContract_NoneWhenNoneExpected.class.getName());
+        List<ContractEvent> events = compUnit.getManifest().getAbi().getEvents();
+        assertThat(events, hasSize(1));
+        assertThat(events.get(0).getName(), is("Hello"));
+        assertThat(events.get(0).getParameters(), hasSize(0));
+    }
+
+    static class EventParameterNamesContract_NoneWhenNoneExpected {
+        @DisplayName("Hello")
+        @EventParameterNames({})
+        public static io.neow3j.devpack.events.Event onHello;
+
+        public static void sayHello() {
+            onHello.fire();
+        }
+    }
+
+    @Test
+    public void testEventParameterNamesWithTooFewNames() {
+        CompilerException thrown = assertThrows(CompilerException.class,
+                () -> new Compiler().compile(EventParameterNamesContract_TooFew.class.getName()));
+        assertThat(thrown.getMessage(), is("Not enough parameter names provided for event Sending."));
+    }
+
+    static class EventParameterNamesContract_TooFew {
+        @DisplayName("Sending")
+        @EventParameterNames({"Amount", "Sender"})
+        public static Event3Args<Integer, Hash160, Hash256> onSending;
+    }
+
+    @Test
+    public void testEventParameterNamesWithNoNames() {
+        CompilerException thrown = assertThrows(CompilerException.class,
+                () -> new Compiler().compile(EventParameterNamesContract_None.class.getName()));
+        assertThat(thrown.getMessage(), is("Not enough parameter names provided for event Call."));
+    }
+
+    static class EventParameterNamesContract_None {
+        @DisplayName("Call")
+        @EventParameterNames({})
+        public static Event3Args<Integer, Hash160, Hash256> onCall;
+    }
```

Comment:
Do we have a test for, e.g., `Event2Args` attribute but no `@EventParameterNames` whatsoever?

This should work, right? Then a default event name will be added. Or am I missing smth?

Reply:
Oh yes, these tests are there already. I didn't remove any tests and previous tests for Event types did not have that annotation.